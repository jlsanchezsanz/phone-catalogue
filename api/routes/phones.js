const phonesRoutes = (app, fs) => {
  const dataPath = './data/phones.json';

  app.get('/phones', (_, res) => {
    fs.readFile(dataPath, 'utf8', (err, data) => {
      setTimeout(() => {
        if (err) {
          throw err;
        }
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.send(JSON.parse(data));
      }, 3000);
    });
  });
};

module.exports = phonesRoutes;
