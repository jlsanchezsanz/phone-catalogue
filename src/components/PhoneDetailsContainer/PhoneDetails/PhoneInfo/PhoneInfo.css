.info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 40%;
}

.info > * {
  margin-bottom: 30px;
}

.info__price {
  color: #326190;
  font-size: 40px;
}

h1 {
  font-size: 24px;
}
