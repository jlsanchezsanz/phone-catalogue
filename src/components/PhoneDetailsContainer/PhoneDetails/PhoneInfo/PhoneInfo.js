import React from 'react';
import PropTypes from 'prop-types';

import { PhonePropType } from '../../../../prop-types/phone.prop-type';
import './PhoneInfo.css';

export default function PhoneInfo(props) {
  return (
    <div className='info'>
      <h1>
        {props.phone.manufacturer} {props.phone.name} {props.phone.color}
      </h1>
      <div className='info__price'>{props.phone.price} â‚¬</div>
      <button type='button' class='btn btn-info'>
        Add to cart
        <img
          class="btn__icon"
          src={require('../../../../assets/cart-plus.svg')}
          alt='Add to cart'
        />
      </button>
    </div>
  );
}

PhoneInfo.propTypes = {
  phone: PropTypes.shape(PhonePropType).isRequired
};
