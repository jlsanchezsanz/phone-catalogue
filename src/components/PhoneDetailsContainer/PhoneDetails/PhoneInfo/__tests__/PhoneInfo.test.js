import React from 'react';
import { shallow } from 'enzyme';

import PhoneInfo from '../PhoneInfo';
import { phonesMock } from '../../../../../mocks/phones.mock';

const setUpShallow = (phone) => {
  const component = shallow(<PhoneInfo phone={phone} />);
  return component;
};

describe('PhoneInfo', () => {
  let component;

  it('should display phone info', () => {
    component = setUpShallow(phonesMock.data[0]);
    expect(component).toMatchSnapshot();
  });
});
