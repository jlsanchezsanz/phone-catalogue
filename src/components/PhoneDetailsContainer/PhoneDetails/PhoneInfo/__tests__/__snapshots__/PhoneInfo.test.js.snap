// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInfo should display phone info 1`] = `
<div
  className="info"
>
  <h1>
    Apple
     
    iPhone 7
     
    black
  </h1>
  <div
    className="info__price"
  >
    769
     â‚¬
  </div>
  <button
    class="btn btn-info"
    type="button"
  >
    Add to cart
    <img
      alt="Add to cart"
      class="btn__icon"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "cart-plus.svg",
        }
      }
    />
  </button>
</div>
`;
